pkg/controller/controller_binding.go

```diff
@@ -87,6 +88,30 @@ func isServiceInstanceCredentialFailed(binding *v1alpha1.ServiceInstanceCredenti
 	return false
 }
 
+func (c *controller) doOrphanMitigation(instance *v1alpha1.ServiceInstance, toUpdate *v1alpha1.ServiceInstanceCredential) error {
+	serviceClass, servicePlan, _, brokerClient, err := c.getServiceClassPlanAndServiceBrokerForServiceInstanceCredential(instance, toUpdate)
+	if err != nil {
+		return err
+	}
+
+	unbindRequest := &osb.UnbindRequest{
+		BindingID:  toUpdate.Spec.ExternalID,
+		InstanceID: instance.Spec.ExternalID,
+		ServiceID:  serviceClass.ExternalID,
+		PlanID:     servicePlan.ExternalID,
+	}
+
```

Comment:
I think we want orphan mitigation to be just as feature-rich as regular deletion. What I did for instances was refactor the deprovision section of `reconcileServiceInstanceDelete(instance)` into a separate `deprovisionServiceInstance(instance)` method that is called in either the orphan mitigation section or the deletion section, and then do a couple of `if instance.Status.OrphanMitigationInProgress` checks for the slight variances. That way all deprovisioning logic is kept in sync (which will be important for bindings as well once async bindings are introduced).